### ISSUE TITLE
newline after or before binary operators

### ISSUE THREAD

**Issue by gnagy** [shyiko/ktlint#197]
Hi,

CONTRIBUTING.md said no opinion is wrong, so here it goes :) I really like all the defaults in ktlint but one rule seems a bit off: newline after binary operators. Where does this rule come from / what is the rationale behind this?

The [kotlin style guide](https://github.com/raywenderlich/kotlin-style-guide) doesn't seem to have a rule for this, and most Java guides seem to advocate breaking before any operators.
https://google.github.io/styleguide/javaguide.html#s4.5.1-line-wrapping-where-to-break
http://cr.openjdk.java.net/~alundblad/styleguide/index-v6.html#toc-wrapping-lines
https://github.com/twitter/commons/blob/master/src/java/com/twitter/common/styleguide.md#indent-style

Even python guys (and Knuth!) seems to prefer breaking before the operator:
https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator

Google's C++ style guide allows both, and seems to prefer breaking before for historical reasons only: https://google.github.io/styleguide/cppguide.html#Boolean_Expressions

Operators on the beginning of the line makes scanning the code easier IMHO, because they are more aligned vertically. Another reason is diffs: if you remove part of an expression, only one line would change in most cases, compared to two or three if line breaks came after the operator. Consider this:

```kotlin
val auth = hasRole("USER")
    && isAuthor(post)
    || hasRole("ADMIN")
```
vs
```kotlin
val auth = hasRole("USER") &&
    isAuthor(post) ||
    hasRole("ADMIN")
```

In the first example, the operators are aligned and so for me easier to read. Removing hasRole('ADMIN') would affect only one line. Perhaps a minor thing, but with such multi-line expressions I find `AND isAuthor post` reads more naturally than `isAuthor post OR linebreak`

So, is there a particular reason why there is a rule to break *after* some operators? Would it be possible to change this default considering most Java style guides seem to suggest otherwise?

**Comment by shyiko** [comment]
Hi. See https://github.com/shyiko/ktlint/issues/163#issuecomment-369418775.
